### PR TITLE
Make sure that ephemerals are cleaned up across gateways.

### PR DESCRIPTION
**Pull request by derekcollison**
Signed-off-by: Derek Collison <derek@nats.io>

/cc @nats-io/core

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -3029,6 +3042,7 @@ func (mset *stream) stateWithDetail(details bool) StreamState {
 	if !details {
 		state.Deleted = nil
 	}
+	state.Consumers += ndc
```
Comment (ripienaar) [inline server/stream.go:3045]: so here Consumers counter will be lets say 10 but we wil not see those in consumer list still which would be confusing don't you think? Definitely welcome more visibility but maybe an additional key on state would be good (ugh), somthing for next release perhaps.

**Review by ripienaar — APPROVED** [review]
LGTM other than the one comment.

File: server/stream.go (hunk 1 above)
Comment (derekcollison) [reply server/stream.go:3045]: Should I make it separate in State?

**Review by kozlovic — APPROVED** [review]
LGTM, but suspect that this will raise questions from user as to the count does not match what one gets from the list.

**Comment by derekcollison** [comment]
Yes changing test and putting back old behavior in terms of stream state as we speak.

**Review by ripienaar — APPROVED** [review]
LGTM

**Pull request merged**